include/deal.II/lac/tensor_product_matrix.h

```diff
@@ -97,41 +97,140 @@ template <typename> class FullMatrix;
+   */
+  mutable Threads::Mutex mutex;
+};
+
+
+
+/**
+ * ... new TensorProductMatrixSymmetricSum using the base class as tensor product
+ * container and interface to arithmetic operations for a generic Number type ...
+ */
+template <int dim, typename Number, int size = -1>
 class TensorProductMatrixSymmetricSum
+  : public TensorProductMatrixSymmetricSumBase<dim,Number,size>
 {
 public:
   /**
    * Constructor.
    */
-  TensorProductMatrixSymmetricSum();
+  TensorProductMatrixSymmetricSum () ;
 
   /**
    * Constructor that is equivalent to the previous constructor and
-   * immediately calling reinit().
+   * immediately calling the corresponding reinit().
    */
-  TensorProductMatrixSymmetricSum(const FullMatrix<Number> &mass_matrix,
-                                  const FullMatrix<Number> &derivative_matrix);
+  TensorProductMatrixSymmetricSum (const std::array<Table<2,Number>, dim> &mass_matrix,
+                                   const std::array<Table<2,Number>, dim> &derivative_matrix) ;
 
   /**
-   * Initializes the matrix to the given mass matrix $M$ and derivative matrix
-   * $A$. Note that the current implementation requires $M$ to be symmetric
-   * and positive definite and $A$ to be symmetric and invertible but not
+   * Constructor that is equivalent to the first constructor and
+   * immediately calling the corresponding reinit().
+   */
+  TensorProductMatrixSymmetricSum (const std::array<FullMatrix<Number>,dim> &mass_matrix,
+                                   const std::array<FullMatrix<Number>,dim> &derivative_matrix) ;
+
+  /**
+   * Constructor that is equivalent to the first constructor and
+   * immediately calling the corresponding reinit().
+   */
+  TensorProductMatrixSymmetricSum (const FullMatrix<Number> &mass_matrix,
+                                   const FullMatrix<Number> &derivative_matrix) ;
+
+  /**
+   * Initializes the tensor product matrix to the given mass matrices $M_0,\ldots,M_{dim}$
+   * and derivative matrices $A_0,\ldots,A_{dim}$.
+   * Note that the current implementation requires each $M_{d}$ to be symmetric
+   * and positive definite and every $A_{d}$ to be symmetric and invertible but not
    * necessarily positive defininte.
    */
-  void reinit (const FullMatrix<Number> &mass_matrix,
-               const FullMatrix<Number> &derivative_matrix);
+  void reinit (const std::array<Table<2,Number>,dim> &mass_matrix,
+               const std::array<Table<2,Number>,dim> &derivative_matrix) ;
 
   /**
-   * Returns the number of rows of this matrix, given by the dim-th power of
-   * the size of the 1D matrices passed to the constructor.
+   * Equivalent to the previous reinit() unless that the mass and derivative
+   * matrices are passed by Table instead of FullMatrix.
    */
-  unsigned int m() const;
+  void reinit (const std::array<FullMatrix<Number>,dim> &mass_matrix,
+               const std::array<FullMatrix<Number>,dim> &derivative_matrix) ;
```

Comment:
Same here: I do not really see why you'd need both versions since `FullMatrix` is a `Table`.

Reply:
Here, it is no problem to have only one constructor/`reinit` that accepts `Table`-typed parameters due to `FullMatrix`'s inheritance.